server/test/unit/com/thoughtworks/go/remote/work/BuildWorkTest.java

```diff
@@ -629,4 +616,31 @@ private void createBuildWorkWithJobPlan(JobPlan jobPlan) throws Exception {
         );
         buildWork = new BuildWork(buildAssignment);
     }
+
+    @Test
+    public void shouldReportEnvironmentVariables() throws Exception {
+        buildWork = (BuildWork) getWork(WITH_ENV_VAR, PIPELINE_NAME);
+
+        buildWork.doWork(agentIdentifier, buildRepository, artifactManipulator, environmentVariableContext, AgentRuntimeInfo.fromAgent(agentIdentifier, "cookie", null), packageAsRepositoryExtension, scmExtension, taskExtension);
+
+        String consoleOut = artifactManipulator.consoleOut();
+
+        assertThat(trimTimeStamp(consoleOut), containsString("[go] setting environment variable 'GO_SERVER_URL' to value 'somewhere-does-not-matter'\n"
+                + "[go] setting environment variable 'GO_TRIGGER_USER' to value ''\n"
+                + "[go] setting environment variable 'GO_PIPELINE_NAME' to value 'pipeline1'\n"
+                + "[go] setting environment variable 'GO_PIPELINE_COUNTER' to value '-2'\n"
+                + "[go] setting environment variable 'GO_PIPELINE_LABEL' to value '100'\n"
+                + "[go] setting environment variable 'GO_STAGE_NAME' to value 'mingle'\n"
+                + "[go] setting environment variable 'GO_STAGE_COUNTER' to value '100'\n"
+                + "[go] setting environment variable 'GO_JOB_NAME' to value 'run-ant'\n"
+                + "[go] setting environment variable 'JOB_ENV' to value 'foobar'\n"
+                + "[go] overriding environment variable 'PATH' with value '/tmp'\n\n"));
+
```

Comment:
Can we make it read like -

```
Setting environment variables -
  - 'FOO'='BAR'
  - 'X'='Y' (overriding)
```

Makes it easy to read and parse and conveys the same information as above paras of text.

Bonus - makes it easy to copy paste into a terminal with minimal edits.